src/Analyser/NodeScopeResolver.php

```diff
@@ -1994,7 +1996,31 @@ static function (?Type $offsetType, Type $valueType, bool $optional) use (&$arra
 				isset($functionReflection)
 				&& in_array($functionReflection->getName(), ['fopen', 'file_get_contents'], true)
 			) {
-				$scope = $scope->assignVariable('http_response_header', new ArrayType(new IntegerType(), new StringType()), new ArrayType(new IntegerType(), new StringType()));
+				$scope = $scope->assignVariable(
+					'http_response_header',
+					TypeCombinator::intersect(
+						new ArrayType(
+							new IntegerType(),
+							TypeCombinator::intersect(
+								new StringType(),
+								new AccessoryNonFalsyStringType(),
+							)
+						),
+						new AccessoryArrayListType(),
+						new NonEmptyArrayType()
+					),
+					TypeCombinator::intersect(
+						new ArrayType(
+							new IntegerType(),
+							TypeCombinator::intersect(
+								new StringType(),
+								new AccessoryNonFalsyStringType(),
+							)
+						),
+						new AccessoryArrayListType(),
+						new NonEmptyArrayType()
```

Comment:
Please try to follow the boy scout rule - always leave the firecamp in a better state than you entered it :) So save the type into a variable and reuse it twice.

Also coding standard check failed.